installtests/service-check.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+# Verify that linux init services correctly handle starting and stopping Jenkins
+set -o
+
+# ARGUMENTS: first argument is the artifact name, jenkins by default if not given
+# Second argument is the port number it will run on for testing
+
+SERVICE_WAIT=5
+MAX_START_WAIT=120
+MAX_STOP_WAIT=45
+
+# Read artifact name as first arg
+if [ -z "$1" ]; then
+    ARTIFACT_NAME=jenkins
+else
+    ARTIFACT_NAME="$1"
+fi
+
+if [ -z "$2" ]; then
+    PORT=8080
+else
+    PORT="$2"
+fi
+
+error_count=0
+
+# Report test results, by looking at status code same as expected
+# Arg 1: test name text
+# Arg 2: test status code output
+# Arg 3: test expected status code to pass
+# Arg 4: command output variable
+function report_test {
+    if [ "$2" -ne "$3" ]; then
+        echo "TEST FAILED - $1 - with status code $2, expected $3"
+        echo "Test command output:"
+        echo "$4"
+        error_count=$((error_count+1))
+    else
+        echo "TEST PASSED - $1  with expected status code $2"
+    fi
+}
+
+# Run a test and look for a specific result
+# Keep running the test every second until it either passes or timout occurs
+function repeatedly_test {
+    local run_command="$1"
+    local expected_status=$2
+    local max_time=$3
+    local test_name="$4"
+    local elapsed=0
+    local exit_code=0
+    local output=""
+
+    while [ $elapsed -lt $max_time ]; do
+        output=$(eval $run_command)
+        exit_code=$?
+        if [ $exit_code -ne "$expected_status" ]; then
+            # Failed, let's wait and retry if time is left
+            elapsed=$((elapsed+1))
+            sleep 1
+        else  # Success!
+            report_test "$test_name" $exit_code $expected_status "$output"
+            return 0
+        fi
+    done
+    report_test "$test_name - with repeated tests on timeout $max_time" $exit_code $expected_status "$output"
+    return 1
+}
+
+# Check if jenkins user is present
+# TODO add check for jenkins group too...
+getent passwd "$ARTIFACT_NAME"
+USER_TEST=$?
+report_test "Verify jenkins user created" $USER_TEST 0 $USER_TEST
```

Comment:
s/jenkins/<code>$USER_TEST</code>? Just to avoid confusions if the username differs